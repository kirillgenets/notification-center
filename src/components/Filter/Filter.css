/* Filter */

.filter {
  box-sizing: border-box;
  width: 241px;

  position: absolute;
  top: 40px;
  z-index: 1;

  background-color: #51A8E7;

  color: #ffffff;
}

.filter::before {
  content: '';

  width: 0; 
  height: 0;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-bottom: 7px solid #51A8E7;

  position: absolute;
  left: 10px;
  top: -7px;
}

.reset {
  width: 100%;
  margin: 8px 0;
  border: none;

  background-color: transparent;

  font-size: 14px;
  color: #ffffff;

  cursor: pointer;
}

.group {
  box-sizing: border-box;
  width: 225px;
  margin: 0 auto;
  margin-bottom: 16px;
  padding: 18px 0 0;
  border: none;
  border-top: 1px solid #ffffff;
}

.row {
  margin-bottom: 8px;
}

.radio {
  display: none;
}

.label {
  padding-left: 28px;

  position: relative;

  font-size: 14px;

  cursor: pointer;
}

.label::before {
  content: '';

  box-sizing: border-box;
  width: 16px;
  height: 16px;
  border-radius: 50%;

  position: absolute;
  left: 0;
  top: 50%;

  background-color: #ffffff;

  transform: translateY(-50%);
}

.radio:checked + .label::before {
  border: 3px solid #ffffff;

  background-color: #4174f7;
}

.submit {
  width: 100%;
  padding: 12px 0;
  border: none;

  background-color:#E3D552;

  font-size: 18px;
  color: #ffffff;

  cursor: pointer;
  transition: 0.3s;
}

.submit:hover {
  opacity: 0.8;
}

.submit:active {
  opacity: 0.6;
}