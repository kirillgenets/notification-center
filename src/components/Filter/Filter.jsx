import React from 'react';
import PropTypes from 'prop-types';
import { CategoryFilters, ReadStatusFilters } from '../../store/constants/constants';
import FilterRadioContainer from '../../containers/FilterRadioContainer';
import styles from './Filter.css';

const Filter = props => {
    const handleFormSubmit = evt => {
        evt.preventDefault();
        
        const filterData = new FormData(evt.target);
        const readStatus = filterData.get('read-status');
        const category = filterData.get('category');

        if (readStatus) {
            props.setReadStatusFilter(readStatus);
        }

        if (category) {
            props.setCategoryFilter(category);
        }
        
        props.onFilterSubmit();
    }

    const handleResetClick = () => {
        props.setReadStatusFilter(ReadStatusFilters.SHOW_ALL);
        props.setCategoryFilter(CategoryFilters.SHOW_ALL);

        props.onFilterSubmit();
    }

    return (
        <form className={styles.filter} onSubmit={handleFormSubmit}>
            <button className={styles.reset} type="button" onClick={handleResetClick}>Reset Filter</button>
            <fieldset className={styles.group}>
                <FilterRadioContainer
                    id="read-radio"
                    name="read-status"
                    value={ReadStatusFilters.SHOW_READ}
                    description="Read"
                />
                <FilterRadioContainer
                    id="unread-radio"
                    name="read-status"
                    value={ReadStatusFilters.SHOW_UNREAD}
                    description="Unread"
                />
            </fieldset>
            <fieldset className={styles.group}>
                <FilterRadioContainer
                    id="critical-radio"
                    name="category"
                    value={CategoryFilters.SHOW_CRITICAL}
                    description="Crit"
                />
                <FilterRadioContainer
                    id="warn-radio"
                    name="category"
                    value={CategoryFilters.SHOW_WARN}
                    description="Warn"
                />
                <FilterRadioContainer
                    id="success-radio"
                    name="category"
                    value={CategoryFilters.SHOW_SUCCESS}
                    description="Success"
                />
                <FilterRadioContainer
                    id="info-radio"
                    name="category"
                    value={CategoryFilters.SHOW_INFO}
                    description="Info"
                />
                <FilterRadioContainer
                    id="error-radio"
                    name="category"
                    value={CategoryFilters.SHOW_ERROR}
                    description="Error"
                />
            </fieldset>
            <button className={styles.submit}>Apply</button>
        </form>
    );
};

Filter.propTypes = {
    setReadStatusFilter: PropTypes.func.isRequired,
    setCategoryFilter: PropTypes.func.isRequired,
    onFilterSubmit: PropTypes.func
}

export default Filter;
