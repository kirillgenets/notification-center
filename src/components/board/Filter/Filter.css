/* Filter */

.filter {
  box-sizing: border-box;
  width: 241px;

  position: absolute;
  top: 40px;
  z-index: 1;

  background-color: #51A8E7;

  color: #ffffff;
}

.filter::before {
  content: '';

  width: 0; 
  height: 0;
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-bottom: 7px solid #51A8E7;

  position: absolute;
  left: 10px;
  top: -7px;
}

.reset {
  width: 100%;
  margin: 8px 0;
  border: none;

  background-color: transparent;

  font-size: 14px;
  color: #ffffff;

  cursor: pointer;
}

.group {
  box-sizing: border-box;
  width: 225px;
  margin: 0 auto;
  margin-bottom: 16px;
  padding: 18px 0 0;
  border: none;
  border-top: 1px solid #ffffff;
}

.submit {
  width: 100%;
  padding: 12px 0;
  border: none;

  background-color:#E3D552;

  font-size: 18px;
  color: #ffffff;

  cursor: pointer;
  transition: 0.3s;
}

.submit:hover {
  opacity: 0.8;
}

.submit:active {
  opacity: 0.6;
}
