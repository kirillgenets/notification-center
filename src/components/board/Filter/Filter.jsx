import React from 'react';
import PropTypes from 'prop-types';
import { CategoryFilters, CompletionStatusFilters } from '../../../store/constants';
import FilterRadioContainer from '../../../containers/FilterRadioContainer';
import styles from './Filter.css';

const Filter = (props) => {
	const handleFormSubmit = (evt) => {
		evt.preventDefault();

		const filterData = new FormData(evt.target);
		const completionStatus = JSON.parse(filterData.get('completion-status'));
		const category = filterData.get('category');

		if (completionStatus) {
			props.setCompletionStatusFilter(completionStatus);
		}

		if (category) {
			props.setCategoryFilter(category);
		}

		props.setCurrentPage(1);

		props.onFilterSubmit();
	};

	const handleResetClick = () => {
		props.setCompletionStatusFilter(CompletionStatusFilters.SHOW_ALL);
		props.setCategoryFilter(CategoryFilters.SHOW_ALL);
		props.setCurrentPage(1);
		props.onFilterSubmit();
	};

	return (
		<form className={styles.filter} onSubmit={handleFormSubmit}>
			<button className={styles.reset} type="button" onClick={handleResetClick}>
				Reset Filter
			</button>
			<fieldset className={styles.group}>
				<FilterRadioContainer
					id="completed-radio"
					name="completion-status"
					value={CompletionStatusFilters.SHOW_COMPLETE}
					description="Completed"
				/>
				<FilterRadioContainer
					id="incomplete-radio"
					name="completion-status"
					value={CompletionStatusFilters.SHOW_INCOMPLETE}
					description="Incomplete"
				/>
			</fieldset>
			<fieldset className={styles.group}>
				<FilterRadioContainer
					id="critical-radio"
					name="category"
					value={CategoryFilters.SHOW_CRITICAL}
					description="Crit"
				/>
				<FilterRadioContainer
					id="warn-radio"
					name="category"
					value={CategoryFilters.SHOW_WARN}
					description="Warn"
				/>
				<FilterRadioContainer
					id="info-radio"
					name="category"
					value={CategoryFilters.SHOW_INFO}
					description="Info"
				/>
				<FilterRadioContainer
					id="error-radio"
					name="category"
					value={CategoryFilters.SHOW_ERROR}
					description="Error"
				/>
				<FilterRadioContainer
					id="debug-radio"
					name="category"
					value={CategoryFilters.SHOW_DEBUG}
					description="Debug"
				/>
			</fieldset>
			<button className={styles.submit}>Apply</button>
		</form>
	);
};

Filter.propTypes = {
	setCompletionStatusFilter: PropTypes.func.isRequired,
	setCategoryFilter: PropTypes.func.isRequired,
	setCurrentPage: PropTypes.func,
	onFilterSubmit: PropTypes.func,
};

export default Filter;
