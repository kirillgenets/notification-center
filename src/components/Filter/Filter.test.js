import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Filter from './Filter';

configure({
  adapter: new Adapter(),
});

it('Filter renders correctly', () => {
  const tree = shallow(
    <Filter
      setCategoryFilter={jest.fn()}
      setReadStatusFilter={jest.fn()}
    />,
  );

  expect(tree).toMatchSnapshot();
});
