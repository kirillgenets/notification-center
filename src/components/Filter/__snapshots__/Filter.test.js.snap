// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <button
    onClick={[Function]}
    type="button"
  >
    Reset Filter
  </button>
  <fieldset>
    <Connect(FilterRadio)
      description="Read"
      id="read-radio"
      name="read-status"
      value="SHOW_READ"
    />
    <Connect(FilterRadio)
      description="Unread"
      id="unread-radio"
      name="read-status"
      value="SHOW_UNREAD"
    />
  </fieldset>
  <fieldset>
    <Connect(FilterRadio)
      description="Crit"
      id="critical-radio"
      name="category"
      value="SHOW_CRITICAL"
    />
    <Connect(FilterRadio)
      description="Warn"
      id="warn-radio"
      name="category"
      value="SHOW_WARN"
    />
    <Connect(FilterRadio)
      description="Success"
      id="success-radio"
      name="category"
      value="SHOW_SUCCESS"
    />
    <Connect(FilterRadio)
      description="Info"
      id="info-radio"
      name="category"
      value="SHOW_INFO"
    />
    <Connect(FilterRadio)
      description="Error"
      id="error-radio"
      name="category"
      value="SHOW_ERROR"
    />
    <Connect(FilterRadio)
      description="Debug"
      id="debug-radio"
      name="category"
      value="SHOW_DEBUG"
    />
  </fieldset>
  <button>
    Apply
  </button>
</form>
`;
